### ISSUE TITLE
代码和论文有点对不上

### ISSUE THREAD

**Issue by bandaidssssss** [xiaozhen228/Bayes-PFL#17]
论文公式7和代码感觉不太一样。代码仅仅使用了elbo的kl散度，没有使用整个elbo。

**Comment by jkchou** [comment]
是的，VPB.py中Context_Prompting.forward_ensemble返回的是 `return text_embeddings, [rec, 0.01 * kl, 0.01 * kl_n, 0.01 * kl_a]`，但在 train.py 中实际上只使用了 KL 项：`loss_dist_reg = loss_reg[1] + loss_reg[2] + loss_reg[3]`，重建损失 rec 并没有被纳入最终的损失计算中。

**Comment by xiaozhen228** [comment]
Hello, thanks for your attention. We have mentioned this point in the loss function section and in the appendix. "The third term, $E_{q_0(\Phi_0)}[\log p(D | \Phi_K)]$, maximizes the log-likelihood of the data, which is approximated by the sum of classification loss (for text-image alignment) and segmentation loss (for text-patch alignment)."

**Comment by bandaidssssss** [comment]
> Hello, thanks for your attention. We have mentioned this point in the loss function section and in the appendix. "The third term, E q 0 ( Φ 0 ) [ log ⁡ p ( D | Φ K ) ] , maximizes the log-likelihood of the data, which is approximated by the sum of classification loss (for text-image alignment) and segmentation loss (for text-patch alignment)."

Thank you for your answer. I didn't notice earlier, my apologies. I'd like to ask if you encountered any instability during the training process. Because in my own code where I used Bayes here, instability occurred initially when I didn't include that returned 'rec'. After adding it, it became much more stable.

**Comment by xiaozhen228** [comment]
@bandaidssssss  I believe that incorporating reconstruction may be a promising direction to explore, as it helps preserve more visual information. When the sampled results are fused with the learnable prompts, an excessively large numerical scale of the sampled vectors can lead to unstable training. Therefore, I introduce a scaling factor, such as (0.01 × KL). In addition, the instability may also stem from the training data itself. By including more auxiliary datasets rather than relying solely on VisA, the model performance can be significantly improved.
